pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1194,21 +1194,26 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 			ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: util.MessageToAny(tlsContext)},
 		}
 	}
-	// convert to transport socket matcher if the mode was auto detected
-	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
-		transportSocket := cluster.TransportSocket
-		cluster.TransportSocket = nil
-		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
-			{
-				Name: "tlsMode-" + model.IstioMutualTLSModeLabel,
-				Match: &structpb.Struct{
-					Fields: map[string]*structpb.Value{
-						model.TLSModeLabelShortname: {Kind: &structpb.Value_StringValue{StringValue: model.IstioMutualTLSModeLabel}},
+
+	// For `Cluster_ORIGINAL_DST` cluster discovery type, the LbPolicy is `Cluster_CLUSTER_PROVIDED` now
+	// Only apply auto mtls to clusters excluding these kind
+	if cluster.LbPolicy != apiv2.Cluster_CLUSTER_PROVIDED {
```

Comment:
Can you add an unit test for this change?

Reply:
+1.

Reply:
ok